src/main/java/org/broadinstitute/hellbender/tools/GetSampleName.java

```diff
@@ -84,9 +99,27 @@ public void onTraversalStart() {
         }
 
         try (final FileWriter fileWriter = new FileWriter(outputSampleNameFile, false)) {
-            fileWriter.write(sampleNames.get(0));
+            final String rawSample = sampleNames.get(0);
+            final String outputSample = urlEncode ? encode(rawSample) : rawSample;
+            fileWriter.write(outputSample);
         } catch (final IOException ioe) {
             throw new UserException("Could not write file.", ioe);
         }
     }
+
+    public static final String encode(final String string) {
```

Comment:
Please move to a utility class. An internal class (Mutect2Engine) should not call into a CLI (see below).  That violates separation of concerns.